### PR TITLE
Thread response groups

### PR DESCRIPTION
**Pull request by armonr**
This changeset addresses responseGroups for threads.

Also a small fix for post missing site_id ed57b7e1d236b9106f6a8b82a681421b359fc0bd

@jogura @jimmordino 

https://wikia-inc.atlassian.net/browse/DIS-84

### CODE REVIEW

File: service/discussion/src/main/java/com/wikia/discussionservice/mappers/HalPostRepresentationMapper.java [hunk 1]
```diff
@@ -43,40 +44,49 @@ public Representation buildRepresentation(int siteId, Post post, UriInfo uriInfo
...
+        .withProperty("lastEditorId", post.getLastEditorId())
+        .withProperty("modificationDate", post.getModificationDate())
+        .withProperty("creationDate", post.getCreationDate())
+        .withProperty("position", post.getPosition())
+        .withProperty("siteId", post.getSiteId())
+        .withProperty("threadId", String.valueOf(post.getThreadId()))
+        .withProperty("title", post.getTitle())
+        .withProperty("latestRevisionId", String.valueOf(post.getLatestRevisionId()))
+        .withProperty("rawContent", String.valueOf(latestRevision.getRawContent()))
+        .withProperty("renderedContent", String.valueOf(latestRevision.getRenderedContent()));
```
Comment (jogura) [inline service/discussion/src/main/java/com/wikia/discussionservice/mappers/HalPostRepresentationMapper.java]: Is `String.valueOf` required for raw and renderedContent?

File: service/discussion/src/main/java/com/wikia/discussionservice/mappers/HalThreadRepresentationMapper.java [hunk 2]
```diff
@@ -63,13 +64,12 @@ public Representation buildRepresentation(int siteId, DiscussionThread thread,
...
               postRepresentationMapper.buildRepresentation(siteId, post, uriInfo));
         }
-        // fallthrough
-      case LARGE:
-      case MEDIUM:
+
         representation.withRepresentation(
             "lastPost",
             postRepresentationMapper.buildRepresentation(siteId, thread.getLastPost(), uriInfo));
-        // fallthrough
```
Comment (jogura) [inline service/discussion/src/main/java/com/wikia/discussionservice/mappers/HalThreadRepresentationMapper.java]: You need this comment to quiet down checkstyles warning.

File: service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceTest.java [hunk 3]
```diff
@@ -199,11 +196,20 @@ protected void getThreadesponseGroup(ResponseGroup responseGroup, ForumThread ex
...
+
     List<Post> placeholderPostList = new ArrayList<>();
     expectedThread.setPosts(placeholderPostList);
 
     resultThread.setPosts(placeholderPostList);
-    assertThat(resultThread, equalTo(expectedThread));
+
+    String actual = resultThread.toString();
+    String expected = expectedThread.toString();
+    assertThat(actual, equalTo(expected));
```
Comment (jogura) [inline service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceTest.java]: I'm going to revert this change. This is masking a problem introduced [on this line](https://github.com/Wikia/pandora/pull/214/files#diff-cdd2a9e101f8e3ccbdd57db3e5444a16R71).

Comment (armonr) [inline service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceTest.java]: Good call. Thanks!

Comment (jogura) [inline service/discussion/src/test/java/com/wikia/discussionservice/resources/ThreadResourceTest.java]: changed here: ab6973cd

**Comment by jogura** [comment]
LGTM :+1:
